rskj-core/src/main/java/co/rsk/rpc/netty/JsonRpcWeb3ServerHandler.java

```diff
@@ -54,26 +52,26 @@ public class JsonRpcWeb3ServerHandler extends SimpleChannelInboundHandler<ByteBu
     private final JsonNodeFactory jsonNodeFactory = JsonNodeFactory.instance;
     private final JsonRpcBasicServer jsonRpcServer;
     private final long defaultTimeout;
+    private final int maxResponseSize;
 
-    public JsonRpcWeb3ServerHandler(Web3 service, List<ModuleDescription> filteredModules, int maxBatchRequestsSize, RskSystemProperties rskSystemProperties) {
-        this.jsonRpcServer = new JsonRpcCustomServer(service, service.getClass(), rskSystemProperties.getRpcModules());
+    public JsonRpcWeb3ServerHandler(Web3 service, JsonRpcWeb3ServerProperties jsonRpcWeb3ServerProperties) {
+        this.jsonRpcServer = new JsonRpcCustomServer(service, service.getClass(), jsonRpcWeb3ServerProperties.getRpcModules());
         List<JsonRpcInterceptor> interceptors = new ArrayList<>();
-        interceptors.add(new JsonRpcRequestValidatorInterceptor(maxBatchRequestsSize));
+        interceptors.add(new JsonRpcRequestValidatorInterceptor(jsonRpcWeb3ServerProperties.getMaxBatchRequestsSize()));
         jsonRpcServer.setInterceptorList(interceptors);
-        jsonRpcServer.setRequestInterceptor(new JsonRpcMethodFilter(filteredModules));
+        jsonRpcServer.setRequestInterceptor(new JsonRpcMethodFilter(jsonRpcWeb3ServerProperties.getRpcModules()));
         jsonRpcServer.setErrorResolver(new MultipleErrorResolver(new RskErrorResolver(), AnnotationsErrorResolver.INSTANCE, DefaultErrorResolver.INSTANCE));
-
-        this.defaultTimeout = rskSystemProperties.getRpcTimeout();
+        this.defaultTimeout = jsonRpcWeb3ServerProperties.getRpcTimeout();
+        this.maxResponseSize = jsonRpcWeb3ServerProperties.getRpcMaxResponseSize();
     }
 
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, ByteBufHolder request) throws Exception {
         ByteBuf responseContent = Unpooled.buffer();
```

Comment:
we might need to make a "final" check of a response size, as response limiter makes an approximate calcs. I believe we could make use of `responseContent.readableBytes()` (similar as it is used [here](https://github.com/rsksmart/rskj/blob/91af660633a23660c58b7146e1da2498a3d8bfbf/rskj-core/src/main/java/co/rsk/rpc/netty/Web3ResultHttpResponseHandler.java#L52)), wdyt?